rspace/src/main/scala/coop/rchain/rspace/history/HistoryInstances.scala

```diff
@@ -557,8 +557,10 @@ object HistoryInstances {
       (start, key, TriePath.empty).tailRecM(traverse)
     }
 
-    override def reset(root: Blake2b256Hash): History[F] =
-      this.copy(root = root, historyStore = CachingHistoryStore(historyStore.historyStore))
+    override def reset(root: Blake2b256Hash): F[History[F]] =
+      Sync[F].delay(
+        this.copy(root = root, historyStore = CachingHistoryStore(historyStore.historyStore))
+      )
```

Comment:
:+1: The correct way to choose between Sync.delay and Applicative.pure to lift a value to F is to use delay when wrapping some expression which needs to be evaluated (function call) and pure when wrapping a single instance of some primitive or complex object (42.pure[F]).